experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/proxies.spec.ts

```diff
@@ -129,3 +129,153 @@ describe("SharedTreeObject", () => {
 		},
 	);
 });
+
+describe("SharedTreeList", () => {
+	describe("inserting nodes created by factory", () => {
+		const _ = new SchemaBuilder({ scope: "test" });
+		const obj = _.object("Obj", { id: _.string });
+		const schema = _.intoSchema(_.list(obj));
+
+		itWithRoot("insertAtStart()", schema, [{ id: "B" }], (list) => {
+			assert.deepEqual(list, [{ id: "B" }]);
+			const newItem = obj.create({ id: "A" });
+			list.insertAtStart([newItem]);
+			assert.deepEqual(list, [{ id: "A" }, { id: "B" }]);
+		});
+
+		itWithRoot("insertAtEnd()", schema, [{ id: "A" }], (list) => {
+			assert.deepEqual(list, [{ id: "A" }]);
+			const newItem = obj.create({ id: "B" });
+			list.insertAtEnd([newItem]);
+			assert.deepEqual(list, [{ id: "A" }, { id: "B" }]);
+		});
+
+		itWithRoot("insertAt()", schema, [{ id: "A" }, { id: "C" }], (list) => {
+			assert.deepEqual(list, [{ id: "A" }, { id: "C" }]);
+			const newItem = obj.create({ id: "B" });
+			list.insertAt(1, [newItem]);
+			assert.deepEqual(list, [{ id: "A" }, { id: "B" }, { id: "C" }]);
+		});
+	});
+
+	describe("removing items", () => {
+		const _ = new SchemaBuilder({ scope: "test" });
+		const schema = _.intoSchema(_.list(_.number));
+
+		itWithRoot("removeAt()", schema, [0, 1, 2], (list) => {
+			assert.deepEqual(list, [0, 1, 2]);
+			list.removeAt(1);
+			assert.deepEqual(list, [0, 2]);
+		});
+
+		itWithRoot("removeRange()", schema, [0, 1, 2, 3], (list) => {
+			assert.deepEqual(list, [0, 1, 2, 3]);
+			list.removeRange(/* start: */ 1, /* end: */ 3);
+			assert.deepEqual(list, [0, 3]);
+		});
+	});
+
+	describe("moving items", () => {
+		describe("within the same list", () => {
+			const _ = new SchemaBuilder({ scope: "test" });
+			const schema = _.intoSchema(_.list(_.number));
+			const initialTree = [0, 1, 2, 3];
+
+			itWithRoot("moveToStart()", schema, initialTree, (list) => {
+				assert.deepEqual(list, [0, 1, 2, 3]);
+				list.moveToStart(/* sourceStart: */ 1, /* sourceEnd: */ 3);
+				assert.deepEqual(list, [1, 2, 0, 3]);
+			});
+
+			itWithRoot("moveToEnd()", schema, initialTree, (list) => {
+				assert.deepEqual(list, [0, 1, 2, 3]);
+				list.moveToEnd(/* sourceStart: */ 1, /* sourceEnd: */ 3);
+				assert.deepEqual(list, [0, 3, 1, 2]);
+			});
+
+			describe("moveToIndex()", () => {
+				function check(index: number, start: number, end: number) {
+					const expected = initialTree.slice(0);
+					// Remove the moved items from [start..end).
+					const moved = expected.splice(start, /* deleteCount: */ end - start);
+					// Re-insert the moved items, adjusting index as necessary.
+					expected.splice(
+						index <= start
+							? index // If the index is <= start, it is unmodified
+							: index >= end
+							? index - moved.length // If the index is >= end, subtract the number of moved items.
+							: start, // If the index is inside the moved window, slide it left to the starting position.
+						/* deleteCount: */ 0,
```

Comment:
I do wish prettier would improve it's indentation of ternaries.

Here is what it looked like before:
```ts
index <= start
    ? index // If the index is <= start, it is unmodified
	: index >= end
	    ? index - moved.length // If the index is >= end, subtract the number of moved items.
		: start,
```